crates/glaredb/src/bin/main.rs

```diff
@@ -51,16 +51,24 @@ fn main() -> Result<()> {
         None => Commands::Local(cli.local_args),
     };
 
-    match &command {
-        Commands::Local(args) => {
-            if let Some(file) = &args.log_file {
-                logutil::init(
-                    cli.verbose,
-                    cli.log_mode.unwrap_or_default().into(),
-                    Some(file),
-                )
-            }
-        }
+    match (&command, cli.log_mode, cli.verbose) {
+        (
+            // User specified a log file, so we should use it.
+            Commands::Local(LocalArgs {
+                log_file: Some(log_file),
+                ..
+            }),
+            _,
+            _,
+        ) => logutil::init(
+            cli.verbose,
+            // Use JSON logging by default when writing to a file.
+            cli.log_mode.unwrap_or(LoggingMode::Json).into(),
+            Some(log_file),
+        ),
```

Comment:
Since we weren't actually running the tests in this crate, there was a regression in https://github.com/GlareDB/glaredb/pull/2187 that went unnoticed.